### ISSUE TITLE
UPDATE #9:1 SET foo.bar = (INSERT INTO...) Syntax Not Working

### ISSUE THREAD

**Issue by big213** [orientechnologies/orientdb#8583]
## Expected behavior  
`UPDATE ... SET foo.bar = (INSERT INTO...)` successfully updates the linkmap with the inserted record (it worked fine in OrientDB 2.2x)

## Actual behavior  
Fails, returns `java.lang.ClassCastException: java.util.ArrayList cannot be cast to com.orientechnologies.orient.core.db.record.OIdentifiable`

## Steps to reproduce  
`CREATE VERTEX V SET foo = {}`
`UPDATE #9:1 SET foo.bar = (INSERT INTO V SET val = 1)`

What does work is if I do:
`CREATE VERTEX V SET id = 1, foo = {}` `//#9:1` 
`CREATE VERTEX V SET id = 2` `//#9:2`
`UPDATE #9:1 SET foo.bar = #9:2` OK

**Comment by big213** [comment]
It appears that the same issue occurs with the `map()` syntax. Consider this example, where I try to create vertex and populate a linkmap in the same query. Maybe this kind of syntax is too much, but should there be a way to do this kind of action in the same query?

`CREATE VERTEX V SET foo = map("key1", (INSERT INTO V SET id = 1), "key2", (INSERT INTO V SET id = 2))`

**Comment by luigidellaquila** [comment]
Hi @big213 

I just pushed a fix to 3.0.x branch, it will be released with v 3.0.9

Thanks

Luigi

**Comment by big213** [comment]
@luigidellaquila Thank you for the quick attention to this issue. However, I just hope that the fix will also address the following query, could you please check?

`CREATE VERTEX V SET foo = map("key1", (INSERT INTO V SET id = 1), "key2", (INSERT INTO V SET id = 2))`

**Comment by luigidellaquila** [comment]
Hi @big213 

Subqueries in that context are not supported by the SQL executor (for now at least)

Thanks

Luigi

**Comment by big213** [comment]
@luigidellaquila thanks for letting me know. I believe it was working as recently as OrientDB 3.0.7, so it is a bit unfortunate that it is no longer supported. How about this query, will it be supported in 3.0.8?

`CREATE VERTEX V SET foo = { "key1": (INSERT INTO V SET id = 1), "key2": (INSERT INTO V SET id = 2) }`

or...

`CREATE VERTEX V SET foo = {}, foo.key1 = (INSERT INTO V SET id=1), foo.key2 = (INSERT INTO V SET id=2)` (this worked in OrientDB 2.2.x)

Basically, I'm looking for a way to create and populate a linkmap in the same query.

**Comment by luigidellaquila** [comment]
Hi @big213 

I'd say it's quite unlikely, it never worked in v 3.0.x.
We will likely support it in the near future.

Thanks

Luigi

**Comment by big213** [comment]
Hi @luigidellaquila 

It looks like OrientDB 3.0.9 has been released, and I tried to test if these queries were functional:

`CREATE VERTEX V SET foo = {}`
`UPDATE #9:1 SET foo.bar = (INSERT INTO V SET val = 1)`

This will work with no defined fields, but if field `foo` has been defined as a LINKMAP on V, it will still return the following error:

`java.lang.ClassCastException: java.util.ArrayList cannot be cast to com.orientechnologies.orient.core.db.record.OIdentifiable`

**Comment by luigidellaquila** [comment]
Hi @big213 

I just pushed another fix, now it should be OK also in that case

Thanks

Luigi
